R/final_size.R

```diff
@@ -44,24 +44,34 @@ final_size <- function(r0 = 2, contact_matrix, demography_vector,
                        prop_initial_infected = 0.001,
                        prop_suscep = NULL) {
 
-  # Check inputs
+  # Check prop_suscep length
   if (is.null(prop_suscep)) {
-    prop_suscep <- rep(1, length(demography_vector))
-  } # Assume fully susceptible if no entry
+    prop_suscep <- 1.0
+  } else {
+    checkmate::assert_numeric(
+      prop_suscep,
+      lower = 0.0, upper = 1.0, finite = TRUE
+    )
+    if (length(prop_suscep) > 1L) {
+      assertthat::assert_that(
+        length(demography_vector) == length(prop_suscep),
+        msg = "demography vector needs to be same size as susceptibility vector"
+      )
```

Comment:
Relating to the conversation on minimizing dependencies, I would either use base R or checkmate here.

Reply:
Removed in https://github.com/epiverse-trace/finalsize/pull/32/commits/66ad999a44cb9e558d5f7b36d12d4018e15b0127, https://github.com/epiverse-trace/finalsize/pull/32/commits/387dc6d201238a87d0953fdb1cf34f16d8ccf8d4